### PR TITLE
Add support for RoundSync as a storage provider

### PR DESCRIPTION
**Pull request by stevesoltys**
This adds support for [RoundSync](https://github.com/newhinton/Round-Sync) as a storage provider, which provides a ton of storage locations for users (since it is an rclone client).

As suggested by [this comment](https://github.com/seedvault-app/seedvault/issues/311#issuecomment-1073212092), this provides a solution for: [Google Drive](https://rclone.org/drive/) (https://github.com/seedvault-app/seedvault/issues/167), [Seafile](https://rclone.org/seafile/) (https://github.com/seedvault-app/seedvault/issues/169), [pCloud](https://rclone.org/pcloud/) (https://github.com/seedvault-app/seedvault/issues/229), [SFTP](https://rclone.org/sftp/) (https://github.com/seedvault-app/seedvault/issues/173), [WebDav](https://rclone.org/webdav/) (https://github.com/seedvault-app/seedvault/issues/317) and https://github.com/seedvault-app/seedvault/issues/196.

Also:
[Proton Drive](https://rclone.org/protondrive/) (#520)
[Mega](https://rclone.org/mega/) (#311)

Possibly more as well?

### CODE REVIEW

File: app/development/scripts/install_app.sh [hunk 1]
```diff
@@ -30,6 +30,8 @@ $ADB push $ROOT_PROJECT_DIR/app/build/outputs/apk/release/app-release.apk /syste
 echo "Installing Seedvault permissions..."
 $ADB push $ROOT_PROJECT_DIR/permissions_com.stevesoltys.seedvault.xml /system/etc/permissions/privapp-permissions-seedvault.xml
 $ADB push $ROOT_PROJECT_DIR/allowlist_com.stevesoltys.seedvault.xml /system/etc/sysconfig/allowlist-seedvault.xml
+$ADB shell am force-stop com.stevesoltys.seedvault
+$ADB shell am broadcast -a android.intent.action.BOOT_COMPLETED
```
Comment (chirayudesai) [inline app/development/scripts/install_app.sh:34]: Unrelated but does this work well?

@t-m-w @grote remember when we were looking into this not working for fdroid?

https://gitlab.com/fdroid/fdroidclient/-/issues/2721

Comment (grote) [inline app/development/scripts/install_app.sh:34]: Unrelated fixes for something?

File: app/src/main/java/com/stevesoltys/seedvault/ui/storage/SafStorageOptions.kt [hunk 2]
```diff
@@ -50,6 +54,49 @@ internal class SafStorageOptions(
...
+                val documentId = cursor.getString(documentIdIndex)
+
+                val childRoot = SafOption(
+                    authority = AUTHORITY_ROUND_SYNC,
+                    rootId = name,
+                    documentId = documentId,
+                    icon = getIcon(context, AUTHORITY_ROUND_SYNC, name, 0),
+                    title = name,
+                    summary = context.getString(R.string.storage_round_sync_summary_prefix) + name,
+                    availableBytes = null,
```
Comment (grote) [inline app/src/main/java/com/stevesoltys/seedvault/ui/storage/SafStorageOptions.kt:89]: The Roundsync provider doesn't expose this? Would be nice to know when dealing with out of space situations.

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -65,6 +65,7 @@
     <string name="storage_fake_drive_summary">Needs to be plugged in</string>
     <string name="storage_available_bytes"><xliff:g example="1 GB" id="size">%1$s</xliff:g> free</string>
     <string name="storage_not_recommended"><xliff:g example="Skynet">%1$s</xliff:g> (Not recommended)</string>
+    <string name="storage_round_sync_summary_prefix" translatable="false">RoundSync Remote: </string>
```
Comment (grote) [inline app/src/main/res/values/strings.xml:68]: Are we sure roundsync isn't translated into other languages? What about `Remote:`?

File: app/src/main/java/com/stevesoltys/seedvault/ui/storage/StorageRootResolver.kt [hunk 4]
```diff
@@ -141,6 +141,10 @@ internal object StorageRootResolver {
                 context.getDrawable(R.drawable.davx5)
             }
 
+            authority == AUTHORITY_ROUND_SYNC -> {
+                context.getDrawable(R.drawable.round_sync)
+            }
+
```
Comment (grote) [inline app/src/main/java/com/stevesoltys/seedvault/ui/storage/StorageRootResolver.kt:147]: Usually, we include an icon only for providers that we want users to be able to install manually during SUW, otherwise take the icon from the package manager directly. Is this the case here? Did I miss the part about opening a market:// link for the app?

**Review by grote — COMMENTED** [review]
Nice work, just some nitpicks above.

File: app/src/main/java/com/stevesoltys/seedvault/ui/storage/SafStorageOptions.kt (hunk 2 above)
Comment (stevesoltys) [reply app/src/main/java/com/stevesoltys/seedvault/ui/storage/SafStorageOptions.kt:89]: Not that I'm aware.

File: app/src/main/java/com/stevesoltys/seedvault/ui/storage/StorageRootResolver.kt (hunk 4 above)
Comment (stevesoltys) [reply app/src/main/java/com/stevesoltys/seedvault/ui/storage/StorageRootResolver.kt:147]: It would save us work to keep this as-is, since this package will one day be available in F-Droid

Comment (grote) [reply app/src/main/java/com/stevesoltys/seedvault/ui/storage/StorageRootResolver.kt:147]: Sometimes vendoring their icon has copyright issues. We had checked those for NextCloud and DavX5.

Comment (chirayudesai) [reply app/src/main/java/com/stevesoltys/seedvault/ui/storage/StorageRootResolver.kt:147]: The app itself is GPLv3

https://roundsync.com/licenses/License.html 

https://github.com/seedvault-app/seedvault/blob/android14/app/src/main/res/drawable/nextcloud.xml mentions AGPL

https://github.com/seedvault-app/seedvault/blob/android14/app/src/main/res/drawable/davx5.xml doesn't mention a license but I vaguely remember looking at this before.

File: app/development/scripts/install_app.sh (hunk 1 above)
Comment (chirayudesai) [reply app/development/scripts/install_app.sh:34]: Wow I had written this 3 weeks ago but it never got posted since GitHub wanted a review to post a comment. :/

Comment (t-m-w) [reply app/development/scripts/install_app.sh:34]: Yep, I do remember having trouble faking a BOOT_COMPLETED broadcast with adb... It apparently never went through.

**Review by grote — APPROVED** [review]

**Comment by avineas** [comment]
When will we get this feature in the stable release? Shouldn't it have arrived with the android 15 update?
